dfastmi/batch/core.py

```diff
@@ -166,7 +168,7 @@ def batch_mode_core(
             
             if version.parse(cfg_version) == version.parse("1"):
                 # version 1
-                [Q, apply_q, q_threshold, time_mi, tstag, T, rsigma, celerity] = ConfigurationChecker.get_levels_v1(reach, config, nwidth)
+                [Q, apply_q, q_threshold, time_mi, tstag, T, rsigma, celerity] = ConfigurationCheckerLegacy()._get_levels(reach, config, nwidth)
```

Comment:
why is a private method called here?

Reply:
should these if else maybe be part of their respective configuration checkers?

Reply:
the factory can just return the configchecker you need and you can just do configchecker.getmestuff()

Reply:
will be tackled in 125

Reply:
Method get_levels will be made public in this issue.

Reply:
get_levels made public in this issue